src/wasm.rs

```diff
@@ -0,0 +1,200 @@
+                ReplSubCommand::Exit => return Ok(serde_json::Value::Null),
+            };
+
+            Ok(result)
+        }
+
+        future_to_promise(async move {
+            run_command_inner(command, wallet, wallet_opts, blockchain, network)
+                .await
+                .map(|v| JsValue::from_serde(&v).expect("Serde serialization failed"))
+                .map_err(|e| e.to_string().into())
+        })
+    }
+}
+
+#[wasm_bindgen]
+#[cfg(feature = "compiler")]
+pub fn compile(policy: String, aliases: String, script_type: String) -> Result<JsValue, String> {
+    fn compile_inner(
+        policy: String,
+        aliases: String,
+        script_type: String,
+    ) -> Result<String, Box<dyn Error>> {
+        use std::collections::HashMap;
+        let aliases: HashMap<String, Alias> = serde_json::from_str(&aliases)?;
+        let aliases: HashMap<String, String> = aliases
+            .into_iter()
+            .map(|(k, v)| (k, v.into_key()))
+            .collect();
+
+        let policy = Concrete::<String>::from_str(&policy)?;
+
+        let descriptor = match script_type.as_str() {
+            "sh" => Descriptor::new_sh(policy.compile()?)?,
+            "wsh" => Descriptor::new_wsh(policy.compile()?)?,
+            "sh-wsh" => Descriptor::new_sh_wsh(policy.compile()?)?,
+            _ => return Err(Box::<dyn Error>::from("InvalidScriptType")),
+        };
+
+        let descriptor: Result<Descriptor<String>, bdk::Error> = descriptor.translate_pk(
+            |key| Ok(aliases.get(key).unwrap_or(key).into()),
+            |key| Ok(aliases.get(key).unwrap_or(key).into()),
+        );
+        let descriptor = descriptor?;
+
+        Ok(descriptor.to_string().into())
+    }
+
+    compile_inner(policy, aliases, script_type)
+        .map(|v| JsValue::from_serde(&v).expect("Serde serialization failed"))
+        .map_err(|e| e.to_string().into())
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(tag = "type", rename_all = "snake_case")]
+#[cfg(feature = "compiler")]
+enum Alias {
+    GenWif,
+    GenExt { extra: String },
+    Existing { extra: String },
+}
+
+#[cfg(feature = "compiler")]
+impl Alias {
+    fn into_key(self) -> String {
+        match self {
+            Alias::GenWif => {
+                let generated: GeneratedKey<bitcoin::PrivateKey, miniscript::Legacy> =
+                    GeneratableDefaultOptions::generate_default().unwrap();
+
+                let mut key = generated.into_key();
+                key.network = Network::Testnet;
+
+                key.to_wif()
+            }
+            Alias::GenExt { extra: path } => {
+                let generated: GeneratedKey<
+                    bitcoin::util::bip32::ExtendedPrivKey,
+                    miniscript::Legacy,
+                > = GeneratableDefaultOptions::generate_default().unwrap();
+
+                let mut xprv = generated.into_key();
+                xprv.network = Network::Testnet;
+
+                format!("{}{}", xprv, path)
```

Comment:
Are you sure there isn't an `m` left behind when printing the derivation path?

Reply:
`path` is just a string coming from user input